### PR TITLE
add unit tests and test data for comment service pashynskyi

### PR DESCRIPTION
**Pull request by pashunskyi**

### CODE REVIEW

**Review by vladtymo — APPROVED** [review]

File: Provis.UnitTests/Core/Services/CommentServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+            _mapperMock.SetupMap(comments, commentsDTO);
+
+            var result = await _commentService.GetCommentListsAsync(taskId);
+
+            result.Should().NotBeNull();
+
+            result.Should().Equal(commentsDTO);
+
+            await Task.CompletedTask;
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/CommentServiceTests.cs]: a lot of enters

File: Provis.UnitTests/Base/TestData/CommentTestData.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        {
+            return new EditCommentDTO()
+            {
+                CommentId = 1,
+                CommentText = "Edited text",
+                WorkspaceId = 1
+            };
+        }
+    }
+}
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Base/TestData/CommentTestData.cs]: For a what reason we need this class?
Why we can't write just With* methods as private methods in test class

File: Provis.UnitTests/Core/Services/CommentServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+            SetupUserWorkspaceGetFirstBySpecAsync(userWorkspaceMock);
+            SetupCommentGetByKeyASync(comment);
+            SetupDeleteCommentAsync(comment);
+            SetupCommentSaveChangesAsync();
+
+            var result = _commentService.DeleteCommentAsync(id, userId, workspaceId);
+
+            result.IsCompleted.Should().BeTrue();
+            result.IsCompletedSuccessfully.Should().BeTrue();
+            await Task.CompletedTask;
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/CommentServiceTests.cs]: Where is await in async call

File: Provis.UnitTests/Core/Services/CommentServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+                _taskRepositoryMock.Object,
+                _commentRepositoryMock.Object,
+                _userWorkspaceRepositoryMock.Object,
+                _workspaceRepositoryMock.Object,
+                _userManagerMock.Object,
+                _mapperMock.Object);
+        }
+
+        [Test]
+        [TestCase("1")]
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/CommentServiceTests.cs]: Why do we need test case if it's only one

File: Provis.UnitTests/Core/Services/CommentServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+            {
+                DateOfCreate = new DateTimeOffset(DateTime.UtcNow, TimeSpan.Zero),
+                UserId = userId
+            };
+            CreateCommentDTO createCommentDTO = new()
+            {
+                CommentText = "Text",
+                TaskId = 1,
+                WorkspaceId = 1
+            };
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/CommentServiceTests.cs]: Let's move it in methods named with*

File: Provis.UnitTests/Core/Services/CommentServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+                CommentText = comment.CommentText,
+                DateTime = comment.DateOfCreate,
+                TaskId = comment.TaskId,
+                UserId = comment.UserId,
+                UserName = comment.User.UserName
+            };
+
+            _mapperMock.Setup(x =>
+               x.Map(It.IsAny<Comment>(), It.IsAny<CommentListDTO>()))
+                   .Returns(commentListDTO);
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/CommentServiceTests.cs]: Let's move it in Setup method

File: Provis.UnitTests/Core/Services/CommentServiceTests.cs [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+            comment.TaskId = createCommentDTO.TaskId;
+
+            _mapperMock.Setup(x => 
+                x.Map(It.IsAny<CreateCommentDTO>(), It.IsAny<Comment>()))
+                    .Returns(comment);
+            SetupCommentAddAsync(comment);
+            SetupCommentSaveChangesAsync();
+
+            comment.Id = 1;
+            comment.User = new User() { UserName = "Username1" };
```
Comment (YevheniiVovk) [inline Provis.UnitTests/Core/Services/CommentServiceTests.cs]: Could you please make your code clean and move out duplication of object creation setups and ect

**Review by herakros — APPROVED** [review]

**Pull request merged**